packages/buidler-etherscan/src/index.ts

```diff
@@ -1,97 +1,268 @@
+The selected network is ${network.name}.
+Common causes:
+  - Wrong address for contract
+  - Wrong network selected or faulty buidler network config`;
+    throw new NomicLabsBuidlerPluginError(pluginName, message);
+  }
+
+  const { Interface } = await import("@ethersproject/abi");
+  const { isABIArgumentLengthError, isABIArgumentTypeError } = await import(
+    "./ABITypes"
+  );
+  const contractInterface = new Interface(contractInformation.contract.abi);
+  let deployArgumentsEncoded;
+  try {
+    deployArgumentsEncoded = contractInterface
+      .encodeDeploy(constructorArguments)
+      .replace("0x", "");
+  } catch (error) {
+    if (isABIArgumentLengthError(error)) {
+      const { contractName, contractFilename } = contractInformation;
+      // TODO: add a list of types and constructor arguments to the error message?
+      const message = `The constructor for ${contractFilename}:${contractName} has ${error.count.types} parameters
+ but ${error.count.values} arguments were provided instead.`;
+      throw new NomicLabsBuidlerPluginError(pluginName, message, error);
+    }
+    if (isABIArgumentTypeError(error)) {
+      const message = `Value ${error.value} cannot be encoded for the parameter ${error.argument}.
+Encoder error reason: ${error.reason}`;
+      throw new NomicLabsBuidlerPluginError(pluginName, message, error);
+    }
+    // Should be unreachable.
+    throw error;
+  }
+
+  // Ensure the linking information is present in the compiler input;
+  compilerInput.settings.libraries = contractInformation.libraryLinks;
+  const compilerInputJSON = JSON.stringify(compilerInput);
+
+  const solcFullVersion = await solcVersionConfig.getLongVersion();
+
+  const { toVerifyRequest, toCheckStatusRequest } = await import(
+    "./etherscan/EtherscanVerifyContractRequest"
+  );
+  const request = toVerifyRequest({
+    apiKey: etherscan.apiKey,
+    contractAddress: address,
+    sourceCode: compilerInputJSON,
+    contractFilename: contractInformation.contractFilename,
+    contractName: contractInformation.contractName,
+    compilerVersion: solcFullVersion,
+    constructorArguments: deployArgumentsEncoded,
+  });
+
+  const { getVerificationStatus, verifyContract, delay } = await import(
+    "./etherscan/EtherscanService"
+  );
+  const response = await verifyContract(etherscanAPIEndpoint, request);
+
+  console.log(
+    `Successfully submitted source code for contract
+${contractInformation.contractFilename}:${contractInformation.contractName} at ${address}
+for verification on etherscan. Waiting for verification result...`
+  );
+
+  const pollRequest = toCheckStatusRequest({
+    apiKey: etherscan.apiKey,
+    guid: response.message,
+  });
+
+  // Compilation is bound to take some time so there's no sense in requesting status immediately.
+  // TODO: make this configurable? The polling interval should be configurable too, within some safety bounds.
```

Comment:
Not worth it IMO. And it's easy to add if someone requests this feature, but I wouldn't add it before that.

@alcuadrado wdyt?

Reply:
I can't picture the situation where someone is about to deploy verify a contract and goes "I'd love to change the time between each polling request"